Tasks/AndroidSigning/androidsigning.ts

```diff
@@ -73,128 +68,77 @@ var zipaligning = (fn: string) => {
-// Configure localization
-tl.setResourcePath(path.join(__dirname, 'task.json'));
-
-// Get files to be signed 
-var filesPattern = tl.getInput('files', true);
-
-// Signing the APK?
-var jarsign: boolean = tl.getBoolInput('jarsign');
+async function run() {
+    try {
+        // Configure localization
+        tl.setResourcePath(path.join(__dirname, 'task.json'));
 
-// Zipaligning the APK?
-var zipalign: boolean = tl.getBoolInput('zipalign');
+        // Get files to be signed 
+        let filesPattern: string = tl.getInput('files', true);
 
-// Resolve files for the specified value or pattern
-if (filesPattern.indexOf('*') == -1 && filesPattern.indexOf('?') == -1) {
-    // No pattern found, check literal path to a single file
-    tl.checkPath(filesPattern, 'files');
+        // Signing the APK?
+        let jarsign: boolean = tl.getBoolInput('jarsign');
 
-    // Use the specified single file
-    var filesList = [filesPattern];
+        // Zipaligning the APK?
+        let zipalign: boolean = tl.getBoolInput('zipalign');
 
-} else {
-    var firstWildcardIndex = function(str) {
-        var idx = str.indexOf('*');
+        // Resolve files for the specified value or pattern
+        let filesToSign: string[] = tl.findMatch(null, filesPattern);
 
-        var idxOfWildcard = str.indexOf('?');
-        if (idxOfWildcard > -1) {
-            return (idx > -1) ?
-                Math.min(idx, idxOfWildcard) : idxOfWildcard;
+        // Fail if no matching files were found
+        if (!filesToSign || filesToSign.length === 0) {
+            throw tl.loc('NoMatchingFiles', filesPattern);
         }
 
-        return idx;
-    }
-
-    // Find app files matching the specified pattern
-    tl.debug('Matching glob pattern: ' + filesPattern);
-
-    // First find the most complete path without any matching patterns
-    var idx = firstWildcardIndex(filesPattern);
-    tl.debug('Index of first wildcard: ' + idx);
-    var findPathRoot = path.dirname(filesPattern.slice(0, idx));
+        for (let file of filesToSign) {
+            if (jarsign) {
+                await jarsigning(file);
+            }
 
-    tl.debug('find root dir: ' + findPathRoot);
-
-    // Now we get a list of all files under this root
-    var allFiles = tl.find(findPathRoot);
-
-    // Now matching the pattern against all files
-    var filesList: string[] = tl.match(allFiles, filesPattern, {matchBase: true});
-
-    // Fail if no matching app files were found
-    if (!filesList || filesList.length == 0) {
-        onError(tl.loc('NoMatchingFiles', filesPattern));
+            if (zipalign) {
+                await zipaligning(file);
+            }
+        }
+    } catch (err) {
+        tl.setResult(tl.TaskResult.Failed, err);
+    } finally {
+        let keystoreFile: string = tl.getTaskVariable('KEYSTORE_FILE_PATH');
+        if (keystoreFile && tl.exist(keystoreFile)) {
+            fs.unlinkSync(keystoreFile);
```

Comment:
It seems like this should be in a "postAndroidSigning.ts" file. Not sure if that exists yet.

Reply:
Post scripts are supported. I will move this there since Luke pointed out a case where this might cause problems. I think it would be a rare issue but very much possible.